state/application.go

```diff
@@ -1005,6 +1005,117 @@ func (a *Application) SetCharm(cfg SetCharmConfig) (err error) {
 	return nil
 }
 
+// UpdateApplicationSeries updates the series for the Application.
+func (a *Application) UpdateApplicationSeries(series string, force bool) (err error) {
+	if a.Series() == series {
+		return jujutxn.ErrNoOperations
+	}
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
+			// If we've tried once already and failed, re-evaluate the criteria.
+			if err := a.Refresh(); err != nil {
+				return nil, errors.Trace(err)
+			}
+			// Exit early if the Application series doesn't need to change.
+			if a.Series() == series {
+				return nil, jujutxn.ErrNoOperations
```

Comment:
Good, this is the right thing to do here. It is treated a as no-op. Hence we should do the same in the CLI (as per my other comment elsewhere).